src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -0,0 +1,1349 @@
+            _renderer->TriggerRedrawAll();
+        }
+    }
+
+    void ControlCore::_updateAntiAliasingMode(::Microsoft::Console::Render::DxEngine* const dxEngine)
+    {
+        // Update DxEngine's AntialiasingMode
+        switch (_settings.AntialiasingMode())
+        {
+        case TextAntialiasingMode::Cleartype:
+            dxEngine->SetAntialiasingMode(D2D1_TEXT_ANTIALIAS_MODE_CLEARTYPE);
+            break;
+        case TextAntialiasingMode::Aliased:
+            dxEngine->SetAntialiasingMode(D2D1_TEXT_ANTIALIAS_MODE_ALIASED);
+            break;
+        case TextAntialiasingMode::Grayscale:
+        default:
+            dxEngine->SetAntialiasingMode(D2D1_TEXT_ANTIALIAS_MODE_GRAYSCALE);
+            break;
+        }
+    }
+
+    // Method Description:
+    // - Update the font with the renderer. This will be called either when the
+    //      font changes or the DPI changes, as DPI changes will necessitate a
+    //      font change. This method will *not* change the buffer/viewport size
+    //      to account for the new glyph dimensions. Callers should make sure to
+    //      appropriately call _doResizeUnderLock after this method is called.
+    // - The write lock should be held when calling this method.
+    // Arguments:
+    // - initialUpdate: whether this font update should be considered as being
+    //   concerned with initialization process. Value forwarded to event handler.
+    void ControlCore::_updateFont(const bool initialUpdate)
+    {
+        const int newDpi = static_cast<int>(static_cast<double>(USER_DEFAULT_SCREEN_DPI) *
+                                            _compositionScaleX);
+
+        // TODO: MSFT:20895307 If the font doesn't exist, this doesn't
+        //      actually fail. We need a way to gracefully fallback.
+        _renderer->TriggerFontChange(newDpi, _desiredFont, _actualFont);
+
+        // If the actual font isn't what was requested...
+        if (_actualFont.GetFaceName() != _desiredFont.GetFaceName())
+        {
+            // Then warn the user that we picked something because we couldn't find their font.
+            // Format message with user's choice of font and the font that was chosen instead.
+            const winrt::hstring message{ fmt::format(std::wstring_view{ RS_(L"NoticeFontNotFound") },
+                                                      _desiredFont.GetFaceName(),
+                                                      _actualFont.GetFaceName()) };
+            auto noticeArgs = winrt::make<NoticeEventArgs>(NoticeLevel::Warning, message);
+            _RaiseNoticeHandlers(*this, std::move(noticeArgs));
+        }
+
+        const auto actualNewSize = _actualFont.GetSize();
+        _FontSizeChangedHandlers(actualNewSize.X, actualNewSize.Y, initialUpdate);
+    }
+
+    // Method Description:
+    // - Set the font size of the terminal control.
+    // Arguments:
+    // - fontSize: The size of the font.
+    void ControlCore::_setFontSize(int fontSize)

```

Comment:
nit: `unsigned int` surely?

Reply:
Shockingly, 

```c#

        String FontFace;
        Int32 FontSize;
```

in IControlSettings. Rather not touch that _obvious_ can of worms